tools/time_tools.go

```diff
@@ -0,0 +1,24 @@
+package tools
+
+import "time"
+
+// IsExpiredAtOrIn returns whether or not the result of calling TimeAtOrIn is
+// "expired" within "until" units of time from now.
+func IsExpiredAtOrIn(now time.Time, until time.Duration, at time.Time, in time.Duration) (time.Time, bool) {
```

Comment:
I'd like to see some golang tests here. Time arithmetic and comparisons always trip me up.